src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,48 @@
+@import "./utils/_variables" ;
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+  width: 706px;
+
+  @each $name, $size in $days {
+    &--start-day-#{$name} &__day:first-child {
+      margin-left: $size;
+    }
+  }
+
+  @for $j from 28 through 31 {
+    &--month-length-#{$j} &__day:nth-child(n + #{$j + 1}) {
+      display: none;
+    }
+  }
+
+  &__day {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    height: $day-size;
+    width: $day-size;
+    border: 1px solid #000;
+    background: #eee;
+    padding: 10px;
+    transition: all ease 0.5s;
+
+    &:hover {
+      background: #dba7ce;
+      cursor: pointer;
+      transform: translateY(-20px);
+    }
+
+    @for $i from 1 through 31 {
+      &:nth-child(#{$i}) {
+        &::before {
+          content: "#{$i}";
+          font-size: $font-size;
+          font-family: $font-family;
```

Comment:
Things that never change like font-size and font family should be outside the loop ;)